packages/subproviders/src/subproviders/trezor.ts

```diff
@@ -0,0 +1,160 @@
+                gasLimit: txData.gas,
+                gasPrice: txData.gasPrice,
+            },
+        });
+
+        if (response.success) {
+            const payload: TrezorSignTxResponsePayload = response.payload;
+            const tx = new EthereumTx(txData);
+
+            // Set the EIP155 bits
+            const vIndex = 6;
+            tx.raw[vIndex] = Buffer.from([1]); // v
+            const rIndex = 7;
+            tx.raw[rIndex] = Buffer.from([]); // r
+            const sIndex = 8;
+            tx.raw[sIndex] = Buffer.from([]); // s
+
+            // slice off leading 0x
+            tx.v = Buffer.from(payload.v.slice(2), 'hex');
+            tx.r = Buffer.from(payload.r.slice(2), 'hex');
+            tx.s = Buffer.from(payload.s.slice(2), 'hex');
+
+            return `0x${tx.serialize().toString('hex')}`;
+        } else {
+            const payload: TrezorResponseErrorPayload = response.payload;
+            throw new Error(payload.error);
+        }
+    }
+    /**
+     * Sign a personal Ethereum signed message. The signing account will be the account
+     * associated with the provided address. If you've added the TrezorSubprovider to
+     * your app's provider, you can simply send an `eth_sign` or `personal_sign` JSON RPC
+     * request, and this method will be called auto-magically.
+     * If you are not using this via a ProviderEngine instance, you can call it directly.
+     * @param data Hex string message to sign
+     * @param address Address of the account to sign with
+     * @return Signature hex string (order: rsv)
+     */
+    public async signPersonalMessageAsync(data: string, address: string): Promise<string> {
+        if (_.isUndefined(data)) {
+            throw new Error(WalletSubproviderErrors.DataMissingForSignPersonalMessage);
+        }
+        assert.isHexString('data', data);
+        assert.isETHAddressHex('address', address);
+        const accountIndex = this._cachedAccounts.indexOf(address);
+        const response: TrezorConnectResponse = await this._trezorConnectClientApi.ethereumSignMessage({
+            path: this._publicKeyPath + `${accountIndex}`,
+            message: data,
+            hex: false,
+        });
+
+        if (response.success) {
+            const payload: TrezorSignMssgResponsePayload = response.payload;
+            return '0x' + payload.signature;
+        } else {
+            const payload: TrezorResponseErrorPayload = response.payload;
+            throw new Error(payload.error);
+        }
+    }
+    /**
+     * eth_signTypedData is currently not supported on Trezor devices.
+     * @param address Address of the account to sign with
+     * @param data the typed data object
+     * @return Signature hex string (order: rsv)
+     */
+    // tslint:disable-next-line:prefer-function-over-method
+    public async signTypedDataAsync(address: string, typedData: any): Promise<string> {
+        throw new Error(WalletSubproviderErrors.MethodNotSupported);
```

Comment:
Let's add a `// TODO` comment here so it's clear that this is a remaining TODO for a future dev.

Reply:
added